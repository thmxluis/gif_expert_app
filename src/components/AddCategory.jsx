import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = ({ target }) => {
    // console.log(target.value);
    setInputValue(target.value);
  };

  const onHangleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onNewCategory(inputValue.trim());

    setInputValue("");
  };

  const onFormSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onNewCategory(inputValue.trim());

    setInputValue("");
  };
  return (
    <form onSubmit={onFormSubmit}>
      <input type="text" placeholder="Buscar gifs" value={inputValue} onChange={(event) => onInputChange(event)} />
    </form>
  );
};
