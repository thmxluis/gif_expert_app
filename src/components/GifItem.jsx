import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
