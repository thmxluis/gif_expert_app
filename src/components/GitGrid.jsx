export const GiftGrid = ({ category }) => {
  const gifs = [1, 2, 3, 4, 5, 6, 7, 8, 9];
  return (
    <>
      <h3>{category}</h3>
      <p>Hola mundo</p>
    </>
  );
};
