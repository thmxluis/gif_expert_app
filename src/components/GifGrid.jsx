import { useState, useEffect } from "react";

import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";

export const GiftGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  const getImages = async () => {
    const gifs = await getGifs(category);
    setImages(gifs);
  };

  useEffect(() => {
    getImages();
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => (
          <GifItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
