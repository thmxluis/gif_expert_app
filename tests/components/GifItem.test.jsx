import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test en <GifItem />", () => {
  const title = "Goku";
  const url = "https://localhost/algo.jpg";
  test("Mostrar Snapshot", () => {
    const { wrapper } = render(<GifItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostar la imagen con el URL y alt de los props", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);

    // Otra forma de hacerlo
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe de mostrar el titulo del componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
