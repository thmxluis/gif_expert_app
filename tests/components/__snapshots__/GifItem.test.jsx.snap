// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifItem /> Mostrar Snapshot 1`] = `undefined`;
